src/OrchardCore.Modules/Orchard.Autoroute/Indexes/AutoroutePartIndex.cs

```diff
@@ -18,7 +18,7 @@ public override void Describe(DescribeContext<ContentItem> context)
                 .Map(contentItem =>
                 {
                     var path = contentItem.As<AutoroutePart>()?.Path;
-                    if (!String.IsNullOrEmpty(path))
+                    if (!String.IsNullOrEmpty(path) && (contentItem.Published || contentItem.Latest))
```

Comment:
You can't don't bother doing an update right now, just change the create and restart setup.
We are not even in Beta, moving parts like this are acceptable.

But as you mention I will soon have to provide a way to update the indexes automatically. I already have some ideas, should be straightforward.